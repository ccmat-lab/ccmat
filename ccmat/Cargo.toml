[package]
name = "ccmat"
version.workspace = true
edition.workspace = true
authors.workspace = true
license.workspace = true
description.workspace = true
repository.workspace = true
homepage.workspace = true
keywords.workspace = true

[dependencies]
ccmat-core = { path="../ccmat-core" }
