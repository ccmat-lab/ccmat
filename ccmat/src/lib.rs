pub use ccmat_core::{lattice_angstrom, sites_frac_coord, CrystalBuilder};
