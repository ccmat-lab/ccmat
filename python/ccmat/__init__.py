from .ccmat import *

__all__ = (

)
