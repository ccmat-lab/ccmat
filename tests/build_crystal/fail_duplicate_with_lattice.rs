use ccmat::{lattice_angstrom, Atom, CrystalBuilder};

fn main() {
    let lattice = lattice_angstrom![
        a = (1.0, 0.0, 0.0),
        b = (0.0, 1.0, 0.0),
        c = (0.0, 0.0, 1.0),
    ];
    let atoms = vec![Atom::new([0.0, 0.0, 0.0], 8)];
    let _ = CrystalBuilder::new()
        .with_lattice(lattice)
        .with_lattice(lattice)
        .with_atoms(&atoms)
        .build()
        .unwrap();
}
