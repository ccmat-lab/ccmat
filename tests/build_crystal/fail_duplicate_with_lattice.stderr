error[E0599]: no method named `with_lattice` found for struct `CrystalBuilder<LatticeSet, AtomsNotSet>` in the current scope
 --> tests/build_crystal/fail_duplicate_with_lattice.rs:8:10
  |
6 |       let _ = CrystalBuilder::new()
  |               ---------------------
  |               |
  |  _____________method `with_lattice` is available on `CrystalBuilder<LatticeNotSet, AtomsNotSet>`
  | |
7 | |         .with_lattice(lattice)
8 | |         .with_lattice(lattice)
  | |_________-^^^^^^^^^^^^
  |
help: there is a method `with_atoms` with a similar name
  |
8 -         .with_lattice(lattice)
8 +         .with_atoms(lattice)
  |
